app/routes.py

```diff
@@ -110,26 +112,51 @@ def categoryHandler(id):
         except Exception as e:
             return str(e), 500
 
-
-    else: 
+    else:
         if (Categories.findById(id) is None):
             return 'fail', 204
         Categories.delete(id)
         return "success", 200
 
 ################################################################################
-#matches 
+# matches
 ################################################################################
-    
+
+
 @cross_origin()
-@app.route("/api/matches", methods = ["GET"])
+@app.route("/api/matches/all", methods=["GET"])
 def getMatches():
-    app.logger.debug('This endpoint was hit!')
+    # app.logger.debug('This endpoint was hit!')
     matches = Matches.query.all()
-    return json.dumps([m.serialize() for m in matches])
+    return json.dumps([m.serialize() for m in matches]), 200
 
-@cross_origin()
-@app.route("/api/match/<id>", methods = ["DELETE", "GET", "PUT"])
+
+@cross_origin
+@app.route('/api/matches', methods=['GET'])
+def getMatchPage():
+    try:
+        # app.logger.debug('hit match pagination')
+        page = request.args.get('page')
+        matchPerPage = 10
+        matches = Matches.query.all()
+        matches = matches[::-1]
+        matchPage = matches[(int(page)-1)*matchPerPage:int(page)*matchPerPage]
+        returnObj = {
+            'metadata':
+            {
+                'recordCount': str(len(matches)),
+                'pageCount': str(int(len(matches)/matchPerPage) + 1),
+                'recordsPerPage': str(matchPerPage),
```

Comment:
nit: how come you need to do all of this type casting?

Reply:
something weird happened with returning obj but seemed to work with strings so I left it as